dali/pipeline/operators/decoder/decoder.cc

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "dali/pipeline/operators/operator.h"
+#include "dali/pipeline/operators/decoder/nvjpeg_decoder_decoupled_api.h"
+#include "dali/pipeline/operators/decoder/host_decoder.h"
+
+namespace dali {
+
+DALI_SCHEMA(Decoder)
+  .DocStr(R"code(Decode images. Implementation will be `nvJPEGDecoder` or `HostDecoder`
+depending on the selected backend (`mixed` and `cpu` respectively).
+Output of the decoder is in `HWC` ordering.)code")
+  .NumInput(1)
+  .NumOutput(1)
+  .AddOptionalArg("output_type",
```

Comment:
What about this extra args that nvJPEG accepts?

Reply:
And slice and crop, and split option for nvJPEG?

Reply:
Please add `.AddParent("nvJPEGDecoder");`

Reply:
Solved